internal/framework/status/updater_test.go

```diff
@@ -508,3 +509,119 @@ var _ = Describe("Updater", func() {
 		})
 	})
 })
+
+func TestUpdateObjStatus(t *testing.T) {
+	// totalFailedAttempts: How many times the fake client fails before returning nil.
+	// allowedFailedAttempts: The total amount of times the function gets run before the
+	//						  status updater stops retrying and errors.
+	tests := []struct {
+		name                  string
+		totalFailedAttempts   int
+		allowedFailedAttempts int
+		expError              bool
+	}{
+		{
+			name:                  "fails when allowedFailedAttempts is less than totalFailedAttempts",
+			totalFailedAttempts:   5,
+			allowedFailedAttempts: 4,
+			expError:              true,
+		},
+		{
+			name:                  "passes when allowedFailedAttempts is greater than totalFailedAttempts",
+			totalFailedAttempts:   5,
+			allowedFailedAttempts: 6,
+			expError:              false,
+		},
+		{
+			name:                  "fails when allowedFailedAttempts is equal to totalFailedAttempts",
+			totalFailedAttempts:   5,
+			allowedFailedAttempts: 5,
+			expError:              true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			g := NewWithT(t)
+			fakeClient := statusfakes.NewFakeClient(test.totalFailedAttempts)
+			upd := status.NewUpdater(status.UpdaterConfig{
+				Client: fakeClient,
+			})
+			if test.expError {
```

Comment:
nit: I think this would be easier to read if you moved the call to UpdateObjStatus above the error check:
```
err := status.UpdateObjStatus()
if test.expErr {
    Expect(err).ToNot(BeNil())
} else {
    Expect(err).To(BeNil())
} 
```

Reply:
Removed tests because we use imported retrying function.